### ISSUE TITLE
Recent change in MouseWatcher.has_mouse() behaviour

### ISSUE THREAD

**Issue by Epihaius** [panda3d/panda3d#363]
After trying out build 96a48a6 of Panda3D, I noticed that `MouseWatcher.has_mouse()` now returns `False` when the mouse moves outside of the Panda window, even when a mouse button is held down. In previous versions (at least up to build 835aab5), as long as a mouse button was held down, `MouseWatcher.has_mouse()` would return `True` when the mouse had left the window.

My use-case is as follows:
to navigate a scene in my project, the user can hold down a mouse button to orbit or pan the view. To implement this, `MouseWatcher.get_mouse()` is called to get the current mouse coordinates, which of course only works if `MouseWatcher.has_mouse()` is `True`.
Previously, it was no problem to drag the mouse outside of the window to keep orbiting or panning as far as the screen allows, but now this is no longer possible, which is a bit annoying.

When running the following code sample, hold down a mouse button while moving the mouse outside of the window:

```
from panda3d.core import *
from direct.showbase.ShowBase import ShowBase


class MyApp(ShowBase):

    def __init__(self):

        ShowBase.__init__(self)
        self.task_mgr.add(self.__check_mouse, "check_mouse")

    def __check_mouse(self, task):

        print("Has mouse:", self.mouseWatcherNode.has_mouse())
        return task.cont


app = MyApp()
app.run()
```

One workaround is to no longer rely on `MouseWatcher.get_mouse()` but on `GraphicsWindow.get_pointer(0)` to get the mouse coordinates, in case this change is intentional.

This happens on Windows 10 with a 64-bit Python 3.6 build of Panda.

**Comment by rdb** [comment]
What about `window.get_pointer(0).in_window` ?

**Comment by rdb** [comment]
Ah, I see what the problem is.  This might be due to a fix to #359 that was recently applied (29a08932ea92bcf0e953994c524cafc1717930b5), which increases the accuracy of pointer position reporting.  However, the new check also has a more strict interpretation of "in_window", ie. simply checking the pointer position against the window rectangle.

I can see how it is useful to retain the `in_window` status while pointer capture is active.  I will check in that change.  It will mean that reading the cursor position may yield an out-of-bounds result, though, so we need to clearly document this behaviour.
